src/producer/index.spec.js

```diff
@@ -76,6 +76,26 @@ describe('Producer', () => {
     )
   })
 
+  test('throws an error for messages with keys that cannot be converted to buffers', async () => {
+    producer = createProducer({ cluster: createCluster(), logger: newLogger() })
+
+    await expect(
+      producer.send({ acks: 1, topic: topicName, messages: [{ key: 1, value: 'value' }] })
+    ).rejects.toThrow(
+      `Invalid message for topic "${topicName}". "key" needs to be convertible to Buffer: {"key":1,"value":"value"}`
+    )
+  })
+
+  test('throws an error for messages with values that cannot be converted to buffers', async () => {
+    producer = createProducer({ cluster: createCluster(), logger: newLogger() })
+
+    await expect(
+      producer.send({ acks: 1, topic: topicName, messages: [{ value: 1 }] })
+    ).rejects.toThrow(
+      `Invalid message for topic "${topicName}". "value" needs to be convertible to Buffer: {"value":1}`
```

Comment:
@Nevon If I'm not wrong, the value can be `null` also. In the case of the compacted topic, you need to send `null` message as a tombstone. https://kafka.apache.org/documentation/#design_compactionbasics

Reply:
I just noticed this was not merged.